### DISCUSSION TITLE
Best practices to re-use non-modal dialog

### DISCUSSION THREAD

**Discussion by andresod** [FantasticFiasco/mvvm-dialogs#244] [Q&A] [answered]
I hope this is the right forum for questions about the usage of mvvm-dialogs.
My goal is to create a window that will be brought to the foreground when the below method is called. I first tried the following
```c#
private void ShowLog()
{
    if (alDlg == null)
    {
        alDlg = new AlarmLogDialogViewModel();
        dialogService.Show(this, alDlg);
    }
    else
    {
        dialogService.Activate(alDlg);
    }
}
``` 

However, that didn't result in the dialog to be shown after minimizing it and I changed the else part to this:
```c#               
    var tmp = dialogService.Close(alDlg);
    dialogService.Show(this, alDlg);
```
Is this the right approach?

Great project by the way.

**Comment by FantasticFiasco** [comment, marked as answer]
It might be, I don't think we've got support to bring minimized windows to the forefront if `Activate` isn't doing it for you. Your solution is a sound one, another alternative could be that you create a new interface inheriting `IDialogService` that has this new method that is capable of doing what you want, and then you will have to create your own class that inherits from both `DialogService` and this new interface you created, and then you can implement your logic there. I haven't tested it myself, it's just an idea.
